src/VirtualDataRepeater.js

```diff
@@ -70,18 +70,7 @@ module.exports = kind({
 
 		this.stabilizeExtent();
 
-		var refresh = this.bindSafely(function() {
-			this.doIt();
-		});
-
-		// refresh is used as the event handler for
-		// collection resets so checking for truthy isn't
-		// enough. it must be true.
-		if(immediate === true) {
-			refresh();
-		} else {
-			this.startJob('refreshing', refresh, 16);
-		}
+		this.doIt();
```

Comment:
Yeah, this effectively means that VDR ignores the `immediate` parameter (although DataRepeater itself still respects it).

Currently there's logic in NewDataList that defers refresh, along the same lines as the reflow-when-showing solution. We could consider moving that to VDR instead.